cpu/stm32/periph/pwm.c

```diff
@@ -53,7 +53,7 @@ uint32_t pwm_init(pwm_t pwm, pwm_mode_t mode, uint32_t freq, uint16_t res)
     dev(pwm)->CR1 = 0;
     dev(pwm)->CR2 = 0;
     for (unsigned i = 0; i < TIMER_CHAN; ++i) {
-        dev(pwm)->CCR[i] = 0;
+        *(uint32_t *)(&dev(pwm)->CCR1 + i) = 0;
```

Comment:
Let's make this look a bit nicer

```suggestion
        TIM_CHAN(pwm, i) = 0;
```

```C
#define TIM_CHAN(t, c) *(uint32_t *)(&dev(t)->CCR1 + c) 
```

Reply:
I added the macro definition in `periph_cpu.h` and now it's used in timer.c, pwm.c and qdec.c.